runtime/alloc.c

```diff
@@ -371,3 +371,25 @@ CAMLexport value caml_alloc_some(value v)
   Field(some, 0) = v;
   CAMLreturn(some);
 }
+
+
+/* For aligned atomic */
+
+CAML_STATIC_ASSERT((sizeof(value)) <= 8);
+
+CAMLprim value caml_alloc_aligned(value val)
+{
+  value var = caml_alloc_shr(7, 0);
+  CAMLassert((var - 8) % Allocation_alignment == 0);
```

Comment:
Shouldn't this depend on Allocation_alignment?

```suggestion
  mlsize_t wo_total_block_size = Allocation_alignment/sizeof(value);
  value var = caml_alloc_shr(wo_total_block_size-1, 0);
  CAMLassert((var - wo_total_block_size) % Allocation_alignment == 0);
```

Reply:
This makes even less sense than the original.  I think `var -8` in the original code was trying to get the address of the block header.  Here, `var - wo_total_block_size` subtracts a number of words from the integer representation of a pointer, and it doesn't mean anything to me.

Reply:
Ah yes, it should have been `sizeof(value)`

Reply:
In other words: 8 means at least two different things in the original code.